c10/CMakeLists.txt

```diff
@@ -52,6 +52,22 @@ target_compile_options(c10 PRIVATE "-DC10_BUILD_MAIN_LIB")
 if(${COMPILER_SUPPORTS_HIDDEN_VISIBILITY})
   target_compile_options(c10 PRIVATE "-fvisibility=hidden")
 endif()
+
+option(C10_USE_IWYU "Use include-what-you-use to clean up header inclusion" OFF)
+if(C10_USE_IWYU)
```

Comment:
Can we add this to the CI too?

Reply:
CI needs to install iwyu and build with the option enabled. It should be an easy task.